christfellowshipchurch/web-app-v2
# Apollos - Rock Person ID Authentication

### About
This PR adds an additional method of authentication that allows a user to pass a Rock Person ID(`rpid`) through a URL, automatically logging the user in when loading the page.

The purpose behind adding this feature is for integration with the new Apollos mobile app. This will allow app users to switch to the website will still staying logged in.

### Test Instructions
First grab your Rock Person ID(`authToken`) by logging in with the following query:
```
mutation authenticate {
  authenticate(identity: "${your_username}", password: "${your_password}"){
    user {
      rock {
        authToken
      }
    }
  }
}
```
Next run the website locally and add the `authToken` as the `rpid` query variable in your URL. See examples below:
- `/home?rpid=${auth_token} `
- `/connect?rpid=${auth_token} `
- `/group/my-group?rpid=${auth_token} `

_Note: Make sure you are NOT LOGGED IN before testing the URL to see the authentication work. (I like to use incognito mode on my browser)._

You should see that the website will automatically log the user in allowing them to access authenticated pages as listed above.

### Screenshots
![image](https://github.com/christfellowshipchurch/web-app-v2/assets/46049974/aecc54ab-396a-4974-8def-f32046b10276)
![image](https://github.com/christfellowshipchurch/web-app-v2/assets/46049974/5451c427-94e3-4cd7-a37d-5db3980f3eab)

### Closes Tickets
[CFDP-2717](https://christfellowshipchurch.atlassian.net/browse/CFDP-2717)

[CFDP-2717]: https://christfellowshipchurch.atlassian.net/browse/CFDP-2717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
